src/main/java/org/terasology/potions/events/BeforeDrinkPotionEvent.java

```diff
@@ -21,119 +21,233 @@
 import gnu.trove.list.array.TFloatArrayList;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.entitySystem.event.ConsumableEvent;
-import org.terasology.potions.HerbEffect;
 import org.terasology.potions.component.PotionComponent;
 import org.terasology.potions.component.PotionEffect;
 
+/**
+ * This event is sent to an entity to allow modification and cancellation of potion drinks.
+ * Note that one event will be sent per potion effect, so individual effects can be cancelled.
+ */
 // TODO: Add extends AbstractValueModifiableEvent?
 public class BeforeDrinkPotionEvent implements ConsumableEvent {
```

Comment:
Done.

I'll add BeforeDrinkPotionEvent (the new one to completely cancel the potion drink event) in the next feature release.